js/app/demo/app.jsx

```diff
@@ -110,12 +113,15 @@ define(['react', 'jsx!editor', 'jsx!messages', 'jsx!fixedCode', 'jsx!configurati
                 });
             }.bind(this))
         },
+        componentDidUpdate: function() {
+            window.localStorage.setItem('linterDemoState', JSON.stringify(this.state));
+        },
         render: function() {
             return (
                 <div className="container editorRow">
                     <div className="row">
                         <div className="col-md-7">
-                            <Editor onChange={this.handleChange} text="var foo = bar;" errors={this.state.messages} />
+                            <Editor onChange={this.handleChange} text={this.initialText} errors={this.state.messages} />
```

Comment:
Passing a static value as the `text` prop only works by virtue of React's virtual DOM diffing: even though changes to the text in the editor cause the parent component to update its state and re-render, the text isn't reset because React identifies that the rendered output from `Editor` is identical and avoids recreating the DOM node, thus preserving the text.

One simple solution is to add a `shouldComponentUpdate` check to the `Editor` component and always return false, since the `text` prop is only used to seed the initial value and any subsequent updates happen outside React's render lifecycle.